examples/regression/2-advanced-analysis/plot_timeseries_enbpi.py

```diff
@@ -0,0 +1,291 @@
+from [2]. It appears that the approach inspired from [2] and ``partial_fit``
+offer higher coverage, but with higher width of PIs and are much slower.
+"""
+import warnings
+
+import numpy as np
+import pandas as pd
+from matplotlib import pylab as plt
+from scipy.stats import randint
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.model_selection import RandomizedSearchCV, TimeSeriesSplit
+
+from mapie.metrics import regression_coverage_score
+from mapie.subsample import BlockBootstrap
+from mapie.time_series_regression import MapieTimeSeriesRegressor
+
+warnings.simplefilter("ignore")
+
+# Load input data and feature engineering
+demand_df = pd.read_csv(
+    "../data/demand_temperature.csv", parse_dates=True, index_col=0
+)
+
+demand_df["Date"] = pd.to_datetime(demand_df.index)
+demand_df["Weekofyear"] = demand_df.Date.dt.isocalendar().week.astype("int64")
+demand_df["Weekday"] = demand_df.Date.dt.isocalendar().day.astype("int64")
+demand_df["Hour"] = demand_df.index.hour
+n_lags = 5
+for hour in range(1, n_lags):
+    demand_df[f"Lag_{hour}"] = demand_df["Demand"].shift(hour)
+
+# Train/validation/test split
+num_test_steps = 24 * 7
+demand_train = demand_df.iloc[:-num_test_steps, :].copy()
+demand_test = demand_df.iloc[-num_test_steps:, :].copy()
+features = ["Weekofyear", "Weekday", "Hour", "Temperature"] + [
+    f"Lag_{hour}" for hour in range(1, n_lags)
+]
+
+X_train = demand_train.loc[
+    ~np.any(demand_train[features].isnull(), axis=1), features
+]
+y_train = demand_train.loc[X_train.index, "Demand"]
+X_test = demand_test.loc[:, features]
+y_test = demand_test["Demand"]
+
+model_params_fit_not_done = False
+if model_params_fit_not_done:
+    # CV parameter search
+    n_iter = 100
+    n_splits = 5
+    tscv = TimeSeriesSplit(n_splits=n_splits)
+    random_state = 59
+    rf_model = RandomForestRegressor(random_state=random_state)
+    rf_params = {"max_depth": randint(2, 30), "n_estimators": randint(10, 100)}
+    cv_obj = RandomizedSearchCV(
+        rf_model,
+        param_distributions=rf_params,
+        n_iter=n_iter,
+        cv=tscv,
+        scoring="neg_root_mean_squared_error",
+        random_state=random_state,
+        verbose=0,
+        n_jobs=-1,
+    )
+    cv_obj.fit(X_train, y_train)
+    model = cv_obj.best_estimator_
+else:
+    # Model: Random Forest previously optimized with a cross-validation
+    model = RandomForestRegressor(
+        max_depth=10, n_estimators=50, random_state=59)
+
+# Estimate prediction intervals on test set with best estimator
+alpha = 0.05
+cv_MapieTimeSeries = BlockBootstrap(
+    n_resamplings=100, length=48, overlapping=True, random_state=59
+)
+
+mapie_plus = MapieTimeSeriesRegressor(
+    model, method="plus", cv=cv_MapieTimeSeries, agg_function="mean", n_jobs=-1
+)
+mapie_enpbi = MapieTimeSeriesRegressor(
+    model, method="plus", cv=cv_MapieTimeSeries, agg_function="mean", n_jobs=-1
+)
+
+gap = 1
```

Comment:
Please comment what this "gap" variable means.

Reply:
I changed the name into step_size